### DISCUSSION TITLE
v1.3.0

### DISCUSSION THREAD

**Discussion by faultables** [kepelet/flo#15] [Announcements]
i plan to treat this as a "major update" despite the version bump being a minor version one. there is no milestone or deadline for this version but i plan to release two "major updates" before 2024 ends. i use this approach because i don't want to mess with app store updates frequently

**Comment by AJV20** [comment]
What are your plans for the home page? Not sure if I'm doing something wrong but mine only shows a static image.

**Reply by faultables** [reply]
maybe something like in-device stats (listening activity, top artists, top albums, top tracks, daily activity, etc) since it will work both offline and online. what do you think?

**Reply by AJV20** [reply]
Love the idea. The biggest issue I have at the moment is that my library is too big so is very hard to find artists. I know the plan is to make the library not be a menu but some libraries are huge

**Reply by faultables** [reply]
thanks! currently i'm working on this library features so users can browse by playlists, artists and songs. there's still a lot of work to do (especially on performance) but i think this is an essential feature flo must have too

since you have a big library, it's worth noting that, at the moment, flo only shows the last 100 albums in your library. sorry about that! this issue should be resolved in 1.3 :)

**Comment by faultables** [comment]
# v1.3.0 is on TestFlight, help me test all exciting features!

> If you're not on TestFlight yet, here's the link to join: https://testflight.apple.com/join/913htqtU

## v1.3.0 Highlights

### Enhancement in Player

Currently, the "background" is somewhat boring; I even considered it a joke. But now, the background has a "translucent background" effect, although I don't use ["average color"](https://developer.apple.com/documentation/musickit/artwork/backgroundcolor) as it should be.

| before (v1.2.0 (3), actual device) | after (v1.3.0 (1), simulator) |
|--------|--------|
| ![IMG_1969](https://github.com/user-attachments/assets/9cc7d3bc-89f1-4d5a-96c1-7b837cbf5422) | ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 01 45 04](https://github.com/user-attachments/assets/e5396195-ba93-44a2-bbfb-1c212e9ec8bb) |

This is far from perfect, but I'm really enjoying (and proud of) the new one!

### Browse by Playlists, Artists, and Songs

The most wanted feature is now live! There are no limits anymore—currently flo limits to only showing the last 100 albums—and now you can browse your collections the way you want:

#### Browse By Artists

| Artist list | Artist detail |
|--------|--------|
| ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 01 53 40](https://github.com/user-attachments/assets/4063a9f2-3be4-4f97-9564-2744388fb8c3) |  ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 01 54 04](https://github.com/user-attachments/assets/96dd4c33-3fdf-426b-b7c7-dddf778a77d7) | 

It shows the artist's discography too!

#### Browse By Playlists

| Playlist list | Playlist detail |
|--------|--------|
| ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 01 57 45](https://github.com/user-attachments/assets/e0603fbe-9ec3-41cd-89a8-2b41900b639f) |  ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 01 58 08](https://github.com/user-attachments/assets/16a7e31a-1286-4c5b-b2cd-76ecf792bc5a)

Currently, flo only streams from playlists online. I'll figure it out later!

#### Browse By Songs

![Simulator Screenshot - iPhone 16 - 2024-11-17 at 02 07 20](https://github.com/user-attachments/assets/567c11ed-d976-4146-a3cd-de457bbd8439)

### Search, search, search

Another most wanted feature! Now you can search anywhere there's a text field.

| in albums | in songs |
|--------|--------|
| ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 02 11 12](https://github.com/user-attachments/assets/035f81d1-3d99-4e7e-aa06-9456dde1efdf) | ![Simulator Screenshot - iPhone 16 - 2024-11-17 at 02 08 57](https://github.com/user-attachments/assets/24c9beb0-e08b-4dee-88c6-6112bc808459) |

### Debug Mode

Since people have various ways of running Navidrome, I can't reproduce every bug. To make troubleshooting easier, I've added "Debug mode" in this version. The feature is very simple: every request that happens in the app will be recorded and stored on your device.

https://github.com/user-attachments/assets/009b8b41-87dc-4c53-810d-ccddf3d1716c

You can inspect requests, responses, and header payloads. Before pasting, make sure you've stripped any sensitive data!

This feature was made possible thanks to [Pulse](https://github.com/kean/Pulse).

### Notable changes

- Some "spring" animations now have shorter durations (from 5s to 2s)
- Album/cover art now will be loaded (and cached!) correctly thanks to [Nuke](https://github.com/kean/Nuke)

## v1.3.0 on App Store

I will submit this to App Store after #17 is resolved. Probably today, probably tomorrow, who knows.

---

if you like flo, please consider:

- Spreading the word!
- Starring this [repo](https://github.com/kepelet/flo/stargazers)
- Giving ratings & reviews on the [App Store](https://apps.apple.com/id/app/flo-navidrome-client/id6529533140)
- Sponsoring my work on [GitHub](https://github.com/sponsors/faultables)

**Reply by TiggsPanther** [reply]
Testflight only seems to show 1.2.0 (3) as the latest available version.

**Reply by faultables** [reply]
Sorry, it's still "waiting for review" for the public tester! We may get notification as soon as the team approves it.

**Reply by faultables** [reply]
it's here!!!

![CleanShot 2024-11-19 at 04 41 28@2x](https://github.com/user-attachments/assets/cf611f8c-815b-475d-aa23-5c7604c5f94f)
